### ISSUE TITLE
Wekan doesnt preserve labels between columns anymore

### ISSUE THREAD

**Issue by kc187r** [wekan/wekan#2404]
Server Setup Information:

    Did you test in newest Wekan?: Yes
    Wekan version: v2.71.0
    Operating System: Lubuntu
    Deployment Method(snap/docker/sandstorm/mongodb bundle/source): snap

Problem description:

    A few updates before i had no issues at all moving cards between columns which have an attached label. Since a few versions (unable to tell which exactly) this feature broke.

    Basically I have a Board with multiple columns, if I add a Label and move the card between these columns the label will be deleted.

    Since it used to work - i assume thats a bug. I also would be really happy if this feature would be working again, so you dont have to redo labels on every card moved.

**Comment by xet7** [comment]
@kc187r 

Please try export and import same board as Wekan JSON. Does it still happen on imported board?

I tried with board that has Swimlanes, but I was unable to make label disappear. I can not duplicate this.

**Comment by kc187r** [comment]
Hello @xet7 ,
i tried following your instructions, without any improvement.

One of my colleagues basically figured out the exact problem:

- All of my board labels are unnamed, when i move one of these cards the label gets deleted
- If I add a named label on the other side, the card keeps this named label. Additional unnamed labels gets deleted too

I am okay with it - if its a design decision that only named labels can be "moved" - but since it worked since a few versions before it still might be an unintended bug.

![_testcard](https://user-images.githubusercontent.com/50586875/57637595-19d93200-75ac-11e9-83a6-5780be12716c.PNG)

this card loses all labels except the blue one on moving it to another column.

If this actually doesn't happen for you - i assume i need to reinstall Wekan, since we use it since 0.6 and maybe there are some issues on the database by now.... Because thats how our wekan handles this things on newly created, newly imported or old boards

with regards!

**Comment by rinnaz** [comment]
Have same issue with versions 2.70-2.72 (snap). AFAIR 2.65 wasn't affected.

**Comment by xet7** [comment]
Hmm, I did not yet test with unnamed labels. I'll test.

**Comment by xet7** [comment]
Moved to here from #2421 

From @savoiebz 

We currently have the same issue on our version installed from sources. The label are losts when card moved. Even in the same list when reordering.

The first line in the changelog file mentions `v2.65 2019-04-24 Wekan release`.

**Comment by EloyID** [comment]
I'm using the Wekan Version v3.11.0 and I have this same bug. When reordering or moving the cards with an unnamed label, the label disappears. Is there already a solution to this problem? 
With regards!

**Comment by ADDAH-temp** [comment]
+1

**Comment by avenqo** [comment]
+1

**Comment by Eisi67** [comment]
The bug is still alive in version 4.10 :-(

**Comment by marc1006** [comment]
@Eisi67 Can you please verify if the bug is still present in 4.13?

**Comment by Eisi67** [comment]
@marc1006: Now it's works (4.13). But now you can't set a name to a label.  :-(

**Comment by marc1006** [comment]
Is there any error on the console? (e. g. web inspector)

**Comment by xet7** [comment]
@marc1006 

For me, it seems that when moving card to another board, all unnamed and named labels are lost. I would think this needs more testing.

**Comment by marc1006** [comment]
> @marc1006
> 
> For me, it seems that when moving card to another board, all unnamed and named labels are lost. I would think this needs more testing.

I know that, that's exactly what I've described in the pull request (https://github.com/wekan/wekan/pull/3154):

> This fixes the issues #2404 and
> #2970 but only if the target board doesn't differ
> from the source board.

**Comment by xet7** [comment]
This was fixed by PR #3154 (commit df9851e2b, 'Copy the labels only if the target board is different'), which explicitly references this issue. In current code (models/cards.js `move`/`copy`) labels are only remapped when the destination board differs from the source board, so moving a card between columns or reordering within the same board no longer touches `labelIds` and unnamed labels are preserved. Closing as fixed; please reopen if you still see labels lost on the latest release.
